packages/shared/src/components/search/SearchBarSuggestion.tsx

```diff
@@ -8,13 +8,14 @@ import {
 } from '../buttons/Button';
 import { AiIcon } from '../icons';
 import { AnalyticsEvent, Origin, TargetType } from '../../lib/analytics';
-import { SearchSession } from '../../graphql/search';
+import { SearchQuestion } from '../../graphql/search';
 import AnalyticsContext from '../../contexts/AnalyticsContext';
 
 export type SuggestionOrigin = Origin.HomePage | Origin.SearchPage;
 
 type SearchBarSuggestionProps = ButtonProps<AllowedTags> & {
-  suggestion?: Partial<SearchSession>;
+  id?: SearchQuestion['id'];
```

Comment:
With this new addition id and prompt actually become mantadory right?
Same actually with origin, I think we can make them mantadory

Reply:
Regarding `id` and `prompt`, as per the comment earlier, this component is being used to show the login CTA when as a search suggestion when the user is not logged in: https://github.com/dailydotdev/apps/blob/8cae5d365838b438d13e206e40e1b7077fa9ccb8/packages/shared/src/components/search/SearchBarSuggestionList.tsx#L29-L34

`origin` can't be mandatory for the same reason as above, but also because when this component is used in `Search History`, we are not passing any `origin`: https://github.com/dailydotdev/apps/blob/8e61ae29f061c60fb69e7774db8804efc0b3ffcb/packages/shared/src/components/search/SearchHistory.tsx#L38-L50

Reply:
From my point of view it's ok if components need props to be optional but your tracking checks do not take it into account so someone could send partial data and then we would get partial data in events that would not be according to spec. 

For example `origin` is always required in events and in this component setup it could be undefined.

Reply:
@capJavert fair point, what would be your suggestion in this case? Should we add a new component used for the login CTA suggestion? 

-- I was also not aware `origin` is always required, based on the [taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit?pli=1#gid=1166511776) it looks like most events don't require it? or maybe there is some default value being set somewhere in the frontend that I missed, if not passed explicitly?

Reply:
To make it specific to this implementation so we can close a PR

> Regarding id and prompt, as per the comment earlier, this component is being used to show the login CTA when as a search suggestion when the user is not logged in

I would create another component for login message

> origin can't be mandatory for the same reason as above, but also because when this component is used in Search History, we are not passing any origin

Per taxonomy sheet there should be `history page` origin that you can add and pass to `SearchHistory` component

Then we can make all props required right and avoid partial event edge cases?

Reply:
@capJavert thank you for the suggestion, I updated based on this discussion, and in the process found a few missing events as well. Take a look at the last commit & let me know your thoughts